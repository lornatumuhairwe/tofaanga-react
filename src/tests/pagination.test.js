import React from 'react';
import renderer from 'react-test-renderer';
import PaginationComp from "../components/Pagination/pagination";
import {mount} from 'enzyme';
import sinon, {spy} from 'sinon';


describe('pagination snapshot', () => {
    const details = [];
    it('matches pagination snapshot', () => {
            const pagination = renderer.create(<PaginationComp details={details} />);
            expect(pagination.toJSON()).toMatchSnapshot();
        },
    );

    it("mounts with only next button when next_url is true", () => {
        const details = {next_url: true, prev_url: false};
        // sinon.spy(PaginationComp.prototype, 'handleSelectNext');
        const pagination = mount(<PaginationComp details={details} />);
        pagination.instance().displaySignup = jest.fn();
        const paginationComponent = pagination.find('Pager');
        expect(paginationComponent.length).toEqual(1);
        paginationComponent.simulate('click');
        // expect(pagination.instance().displaySignup).toHaveBeenCalled();

    });

    it("mounts with only prev button when prev_url is true", () => {
        const details = {next_url: false, prev_url: true};
        // sinon.spy(PaginationComp.prototype, 'handleSelectNext');
        const pagination = mount(<PaginationComp details={details} />);
        pagination.instance().displaySignup = jest.fn();
        const paginationComponent = pagination.find('Pager');
        const paginationItem = paginationComponent.find('Pager.Item');
        expect(paginationComponent.length).toEqual(1);
        paginationComponent.simulate('click');
        console.log(paginationItem);
        // expect(pagination.instance().displaySignup).toHaveBeenCalled();

    });
});
