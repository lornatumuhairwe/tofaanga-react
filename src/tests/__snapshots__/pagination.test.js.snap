// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pagination snapshot matches pagination snapshot 1`] = `null`;
